riscv/src/code_gen.rs

```diff
@@ -964,10 +963,14 @@ fn process_instruction<A: InstructionArgs>(instr: &str, args: A) -> Result<Vec<S
                 vec![
                     format!("to_signed {}, {};", r1.addr(), tmp1.addr()),
                     // tmp2 is 1 if tmp1 is non-negative
-                    format!("is_positive {}, 0, 1, {};", tmp1.addr(), tmp2.addr()),
+                    format!(
+                        "is_diff_greater_than {}, 0, -1, {};",
```

Comment:
Nice, that makes much more sense now, to test if it's non negative you test if it's greater than -1.